File: src/main/java/engine/core/util/utilgeometry/PointF.java
package engine.core.util.utilgeometry;

@SuppressWarnings("WeakerAccess")
public class PointF {
    public final float x;
    public final float y;

    public PointF(){
        this.x = 0f;
        this.y = 0f;
    }

    public PointF(final float x, final float y){
        this.x = x;
        this.y = y;
    }

    public String toString(){
        return "(x: " + x + " ; " + "y: " + y + " )";
    }
}

File: src/main/java/engine/util/geometry/PointF.kt
package engine.util.geometry

class PointF(val x: Float, val y: Float) {

    override fun toString() = "(x: $x ; y: $y )"

    override fun equals(other: Any?) =
            if (other is PointF) {
                x == other.x && y == other.y
            } else {
                false
            }
}
